Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -796,11 +795,79 @@ namespace xr
+            int32_t size{};
+            ArTrackableList_getSize(session, trackableList, &size);
+            for (int i = 0; i < size; i++)
+            {
+                // Get the plane.
+                ArPlane* planeTrackable{};
+                {
+                    ArTrackable* trackable{};
+                    ArTrackableList_acquireItem(session, trackableList, i, &trackable);
+                    planeTrackable = reinterpret_cast<ArPlane*>(trackable);
+                }
+
+                // Check if this plane has been subsumed. If so skip it as we are about to delete this plane.
+                ArPlane* subsumingPlane{};
+                ArPlane_acquireSubsumedBy(session, planeTrackable, &subsumingPlane);
+                if (subsumingPlane != nullptr)
+                {
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(planeTrackable));
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(subsumingPlane));
+                    continue;
+                }
+
+                // Get the center pose.
+                float rawPose[7]{};
+                ArPlane_getCenterPose(session, planeTrackable, tempPose);
+                ArPose_getPoseRaw(session, tempPose, rawPose);
+
+                // Dynamically allocate the polygon vector, and fill it in.
+                int32_t polygonSize;
+                ArPlane_getPolygonSize(session, planeTrackable, &polygonSize);
+                planePolygonBuffer.clear();
+                planePolygonBuffer.resize(polygonSize);
+                ArPlane_getPolygon(session, planeTrackable, planePolygonBuffer.data());
+
+                // Update the existing plane if it exists, otherwise create a new plane, and add it to our list of planes.
+                auto planeIterator = planeMap.find(planeTrackable);
+                if (planeIterator != planeMap.end())
+                {
+                    UpdatePlane(updatedPlanes, Planes[planeIterator->second], rawPose, planePolygonBuffer, polygonSize);
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(planeTrackable));
+                }
+                else
+                {
+                    // This is a new plane, create it and initialize its values.
+                    Planes.emplace_back();
+                    auto& plane = Planes.back();
+                    plane.ID = nextPlaneID++;
+                    planeMap.insert({planeTrackable, plane.ID});
+                    UpdatePlane(updatedPlanes, plane, rawPose, planePolygonBuffer, polygonSize);
+                }
+            }
+        }
+
     private:
         bool isInitialized{false};
         bool sessionEnded{false};
         std::vector<ArTrackable*> frameTrackables{};
         std::vector<ArAnchor*> arCoreAnchors{};
+        std::vector<float> planePolygonBuffer{};
+        std::unordered_map<ArPlane*, Frame::Plane::Identifier> planeMap{};
+        Frame::Plane::Identifier nextPlaneID = 0;
```

Comment:
I think you can eliminate this variable if you make the plane IDs set themselves.